__tests__/apiTests.test.js

```diff
@@ -0,0 +1,61 @@
+const request = require("supertest")
+const app = require("../app/app.js")
+const db = require("../db/connection.js")
+const seed = require("../db/seeds/seed.js")
+const data = require("../db/data/test-data/index.js")
+
+
+
+afterAll(()=>{
+    db.end()
+})
+beforeEach(()=>{
+    return seed(data)
+})
+
+describe("GET /*",()=>{
+    test("when given an endpoint that doesnt exist return an error message",()=>{
+        return request(app)
+        .get("/John")
+        .expect(404)
+            .then(({body})=>{
+                const err = body.msg
+                expect(err).toBe("Sorry can't find that!!")
+            })
+    })
+})
+
+
+describe("GET /api/topics",()=>{
+    test("get status code 200, and return all from the topic table",()=>{
+        return request(app)
+            .get("/api/topics")
+            .expect(200)
+                .then(({body})=>{
+                    const topics = body.topics
+
+                    expect(topics).toEqual(data.topicData)
+                    expect(topics.length).toBe(data.topicData.length)
+                    expect(topics.length).toBeGreaterThan(0)
```

Comment:
As we are checking that the length is equal to the length of the topic data, we don't need to check that it's greater than 0 :)